File: 110-balanced-binary-tree/balanced-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isBalanced = function(root) {

    let isBalanced = true;

    const _dfs = (node) => {
        if (!node) return 0;

        const leftHeight = _dfs(node.left);
        const rightHeight = _dfs(node.right);

        if (Math.abs(leftHeight - rightHeight) >= 2) {
            isBalanced = false;
        };

        return 1 + Math.max(leftHeight, rightHeight)
    }

    _dfs(root)
    return isBalanced;
};

File: 110-balanced-binary-tree/balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        isBalanced = [True]

        def dfs(node):
            if not node: 
                return 0

            leftHeight = dfs(node.left)
            rightHeight = dfs(node.right)

            if abs(leftHeight - rightHeight) >= 2: 
                isBalanced[0] = False

            return 1 + max(leftHeight, rightHeight)

        dfs(root)
        return isBalanced[0]
        